p2p/server.go

```diff
@@ -105,6 +104,10 @@ func (s *peer) Start(ctx context.Context, wg *sync.WaitGroup, startupErr chan<-
 	}
 
 	<-ctx.Done()
+
+	if err := s.host.Close(); err != nil {
```

Comment:
@mikiquantum This is the closest thing that we have, related to the below question.